hoodie-client/src/main/java/com/uber/hoodie/config/HoodieWriteConfig.java

```diff
@@ -258,23 +258,45 @@ public double getBloomFilterFPP() {
   }
 
   public String getHbaseZkQuorum() {
-    return props.getProperty(HoodieIndexConfig.HBASE_ZKQUORUM_PROP);
+    return props.getProperty(HoodieHBaseIndexConfig.HBASE_ZKQUORUM_PROP);
   }
 
   public int getHbaseZkPort() {
-    return Integer.parseInt(props.getProperty(HoodieIndexConfig.HBASE_ZKPORT_PROP));
+    return Integer.parseInt(props.getProperty(HoodieHBaseIndexConfig.HBASE_ZKPORT_PROP));
   }
 
   public String getHbaseTableName() {
-    return props.getProperty(HoodieIndexConfig.HBASE_TABLENAME_PROP);
+    return props.getProperty(HoodieHBaseIndexConfig.HBASE_TABLENAME_PROP);
   }
 
   public int getHbaseIndexGetBatchSize() {
-    return Integer.valueOf(props.getProperty(HoodieIndexConfig.HBASE_GET_BATCH_SIZE_PROP));
+    return Integer.valueOf(props.getProperty(HoodieHBaseIndexConfig.HBASE_GET_BATCH_SIZE_PROP));
   }
 
   public int getHbaseIndexPutBatchSize() {
-    return Integer.valueOf(props.getProperty(HoodieIndexConfig.HBASE_PUT_BATCH_SIZE_PROP));
+    return Integer.valueOf(props.getProperty(HoodieHBaseIndexConfig.HBASE_PUT_BATCH_SIZE_PROP));
+  }
+
+  public Boolean getHbaseIndexPutBatchSizeAutoCompute() {
+    return Boolean.valueOf(props.getProperty(HoodieHBaseIndexConfig.HBASE_PUT_BATCH_SIZE_AUTO_COMPUTE_PROP));
+  }
+
+  /**
+   * Fraction of the global share of QPS that should be allocated to this job.
+   * Let's say there are 3 jobs which have input size in terms of number of rows
+   * required for HbaseIndexing as x, 2x, 3x respectively. Then this fraction for
+   * the jobs would be (0.17) 1/6, 0.33 (2/6) and 0.5 (3/6) respectively.
+   */
+  public float getHbaseIndexQPSFraction() {
```

Comment:
why can't these configs also go into HBaseIndexConfig?

Reply:
@vinothchandar I drew a parallel from HoodieCompactionConfig/HoodieIndexConfig, which have builder/setters, but getters for their configs are actually in HoodieWriteConfig.

Reply:
@vinothchandar There are 2 ways of achieving above (moving hbase specific getters to HbaseIndexConfig). Tell me which one you prefer
1) Continue storing all string props in HoodieIndexConfig and HoodieWriteConfig. Fetch props from HoodieWriteConfig and instantiate HoodieHbaseIndexConfig in HbaseIndex, like below
```
  public HBaseIndex(HoodieWriteConfig config) {
    super(config);
    this.hBaseIndexConfig = new HoodieHBaseIndexConfig.Builder().fromProperties(config.getProps()).build();
```
2) Storing an instance of HoodieIndexConfig in HoodieHbaseIndexConfig and an instance of HoodieIndexConfig in HoodieWriteConfig. And use as writeConfig.getIndexConfig().getHbaseIndexConfig().getHbaseZkQuorum() etc. This does not seem neat.

Reply:
lets leave it as-is for now.